pynapple/process/decoding.py

```diff
@@ -134,64 +134,81 @@ def decode_2d(tuning_curves, group, feature, ep, bin_size, xy):
 	    If indexes don't match between tuning_curves and group
 
 	"""
-	if type(group) is dict:
-		newgroup = nap.TsGroup(group, time_support = ep)
-	elif type(group) is nap.TsGroup:
-		newgroup = group.restrict(ep)
-	else:
-		raise RuntimeError("Unknown format for group")
-
-	if len(tuning_curves) != len(newgroup):
-		raise RuntimeError("Different shapes for tuning_curves and group")
-
-	if not np.all(np.array(list(tuning_curves.keys())) == np.array(newgroup.keys())):
-		raise RuntimeError("Difference indexes for tuning curves and group keys")	
+    
+    if type(group) is dict:
+        newgroup = nap.TsGroup(group, time_support = ep)
+        numcells = len(newgroup)
+    elif type(group) is nap.TsGroup:
+        newgroup = group.restrict(ep)
+        numcells = len(newgroup)
+    elif type(group) is nap.TsdFrame:
+        newgroup = group.restrict(ep)
+        numcells = len(group.columns)
+    else:
+        raise RuntimeError("Unknown format for group")
+
+    if len(tuning_curves) != numcells:
+        raise RuntimeError("Different shapes for tuning_curves and group")
+
+    if not np.all(np.array(list(tuning_curves.keys())) == np.array(newgroup.keys())):
+        raise RuntimeError("Difference indexes for tuning curves and group keys")	
 
 	# Bin spikes
-	count = newgroup.count(bin_size, ep, 's')
+    if type(newgroup) is not nap.TsdFrame:
+        count = newgroup.count(bin_size, ep, 's')
+    else:
+        #Spikes already "binned" with continuous TsdFrame input
+        count = newgroup
 
 	# Occupancy
-	binsxy = []
-	for i in range(len(xy)):
-		diff = np.diff(xy[i])
-		bins = xy[i][:-1] - diff/2
-		bins = np.hstack((bins, [bins[-1]+diff[-1],bins[-1]+2*diff[-1]])) # assuming the size of the last 2 bins is equal
-		binsxy.append(bins)
-
-	occupancy, _, _ = np.histogram2d(
-		feature.iloc[:,0],
-		feature.iloc[:,1],
-		[binsxy[0], binsxy[1]])
-	occupancy = occupancy.flatten()
+    if feature is None:
+        #For conditions (e.g. sleep) where you don't have, or don't want to 
+        #bias by behavioral occupancy for decoding
+        occupancy = np.ones(1)
+        #TODO: Assuming features are x and y.... this is hacky
+        feature = nap.TsdFrame(1,columns = ('x','y'))
```

Comment:
I am missing the idea here :thinking:

Reply:
It was a hack so that the column labels got passed through at the end. Maybe the better thing to do would be if there's no Feature passed in, it replaces with a fake feature TsdFrame with uniform occupancy?